service/src/LevelZeroIOGroup.cpp

```diff
@@ -972,6 +1006,21 @@ namespace geopm
                                 " GPU_CHIP domain " + std::to_string(domain_idx),
                                 GEOPM_ERROR_INVALID, __FILE__, __LINE__);
             }
+
+            try {
+                // Currently only the levelzero compute domain control is supported.
+                // As new controls are added they should be included
+                m_perf_factor.at(domain_idx) =
+                    m_levelzero_device_pool.performance_factor(GEOPM_DOMAIN_GPU_CHIP,
+                                                               domain_idx,
+                                                               geopm::LevelZero::M_DOMAIN_COMPUTE);
+            }
+            catch (const geopm::Exception &ex) {
+                throw Exception("LevelZeroIOGroup::" + std::string(__func__) + ": "
```

Comment:
``geopm::Exception()``

Note: the exception handling here is truncating the error message generated by the underlying API (LevelZeroDevicePool, LevelZero). If we don't have a convenient mechanism for nesting and reporting exceptions, maybe append the error message from the underlying exception to this one. That way, besides reporting that the operation failed, you provide the reason why.